node/impl/full/state.go

```diff
@@ -489,13 +489,11 @@ func (a *StateAPI) StateMarketDeals(ctx context.Context, tsk types.TipSetKey) (m
 
 		var s market.DealState
 		if err := sa.Get(ctx, i, &s); err != nil {
-			if err != nil {
-				if _, ok := err.(*amt.ErrNotFound); !ok {
-					return xerrors.Errorf("failed to get state for deal in proposals array: %w", err)
-				}
-
-				s.SectorStartEpoch = -1
+			if _, ok := err.(*amt.ErrNotFound); !ok {
```

Comment:
Use: `errors.As`
```
var notFound *amt.ErrNotFound
if errors.As(err, notFound) {

}
```

Reply:
but `if _, ok := err.(*amt.ErrNotFound); !ok {` is one line :c (and arguably more readable)

Reply:
It will break if anything along the call chain uses `errors.Errorf("bla bla bla: %w")`